### PR TITLE
Remove FSTMutationQueue

### PR DESCRIPTION
**Pull request by gsoltis**
Ports remaining test usages and production usages of `FSTMutationQueue` over to C++ `MutationQueue`. Deletes `FSTMutationQueue` and implementors.

### CODE REVIEW

File: Firestore/Source/Local/FSTLocalStore.mm [hunk 1]
```diff
@@ -124,33 +125,38 @@ - (void)start {
...
+                                                               mutationQueue:_mutationQueue];
 
     // Union the old/new changed keys.
     DocumentKeySet changedKeys;
-    for (NSArray<FSTMutationBatch *> *batches in @[ oldBatches, newBatches ]) {
-      for (FSTMutationBatch *batch in batches) {
-        for (FSTMutation *mutation in batch.mutations) {
-          changedKeys = changedKeys.insert(mutation.key);
-        }
+    for (FSTMutationBatch *batch : oldBatches) {
```
Comment (rsgowman) [inline Firestore/Source/Local/FSTLocalStore.mm]: Does this work:
```
for (const vector<FSTMutationBatch*>& batches : { oldBatches, newBatches} ) {
  for (FSTMutationBatch *batch : batches) {
    for (FSTMutation *mutation in batch.mutations) {
      changedKeys = changedKeys.insert(mutation.key);
    }
  }
}
```
?

Comment (gsoltis) [reply Firestore/Source/Local/FSTLocalStore.mm]: Yup, done.

File: Firestore/Source/Local/FSTLocalDocumentsView.mm [hunk 2]
```diff
@@ -67,16 +71,16 @@ - (instancetype)initWithRemoteDocumentCache:(RemoteDocumentCache *)remoteDocumen
 }
 
 - (nullable FSTMaybeDocument *)documentForKey:(const DocumentKey &)key {
-  NSArray<FSTMutationBatch *> *batches =
-      [self.mutationQueue allMutationBatchesAffectingDocumentKey:key];
-  return [self documentForKey:key inBatches:batches];
+  std::vector<FSTMutationBatch *> batches =
+      _mutationQueue->AllMutationBatchesAffectingDocumentKey(key);
+  return [self documentForKey:key inBatches:std::move(batches)];
```
[the remark this answers is not in the document]
Comment (gsoltis) [reply Firestore/Source/Local/FSTLocalDocumentsView.mm]: Used const reference.

File: Firestore/Source/Local/FSTLocalDocumentsView.mm [hunk 3]
```diff
@@ -86,13 +90,13 @@ - (nullable FSTMaybeDocument *)documentForKey:(const DocumentKey &)key
 // Returns the view of the given `docs` as they would appear after applying all
 // mutations in the given `batches`.
 - (MaybeDocumentMap)applyLocalMutationsToDocuments:(const MaybeDocumentMap &)docs
-                                       fromBatches:(NSArray<FSTMutationBatch *> *)batches {
+                                       fromBatches:(std::vector<FSTMutationBatch *>)batches {
```
[the remark this answers is not in the document]
Comment (gsoltis) [reply Firestore/Source/Local/FSTLocalDocumentsView.mm]: done.

**Pull request merged**